skipper-ingress: explicit healthcheck routes on the pod level

ponimas opened the pull request:
Skipper provides two flags to control kubernetes healthcheck routes:
`-kubernetes-healthcheck` (true by default) that enables automatic healthcheck routes
and `-reverse-source-predicate` to select source predicate type.

Currently when routesrv is enabled (default state) it adds
automatic healthcheck routes and appends default filters to
all routes including healthcheck routes.

This change defines healthcheck routes in skipper pods to avoid adding default
filters to them and enable further modifications. 


The automated healthcheck routes are still enabled on the routesrv level, to 
make it possible to deploy the change and avoid the situation when the skipper
pod without inlined routes consumes data from the routerrv with healthchecks
 disabled. These routes are disabled in the [following PR](https://github.com/zalando-incubator/kubernetes-on-aws/pull/9241). 

When routesrv is disabled though skipper will add default filters
to explicitly-defined healthcheck routes like it happens currently.
In this case default filters should be configured to not hinder
healthchecks.

AlexanderYastrebov committed:
skipper-ingress: explicit healthcheck routes

Skipper provides two flags to control kubernetes healthcheck routes:
`-kubernetes-healthcheck` (true by default) that enables automatic healthcheck routes
and `-reverse-source-predicate` to select source predicate type.

Currently when routesrv is enabled (default state) it adds
automatic healthcheck routes and appends default filters to
all routes including healthcheck routes.

This change disables automatic healthcheck routes and instead
explicitly defines them in skipper pods to avoid adding default
filters to them and enable further modification.

When routesrv is disabled though skipper will add default filters
to explicitly-defined healthcheck routes like it happens currently.
In this case default filters should be configured to not hinder
healthchecks.

Signed-off-by: Alexander Yastrebov <alexander.yastrebov@zalando.de>

ponimas committed:
fix: Add IPv6 EKS subnets to to Skipper healthz routes

ponimas committed:
fix: Remove changes from routesrv

We should have k8s healthchecks in routesrv to pass e2e load
tests. Tests are started with the tip of the dev branch and later the
new configuration is applied. There is a timespan when old ingress
pods consume routes from the new routesev. If healthcheck is false they
will have no healthz endpoint for some time.

Signed-off-by: Aleksandr Ponimaskin <aleksandr.ponimaskin@zalando.de>

shyamz-22 commented:
👍🏽

mikkeloscar commented on cluster/manifests/skipper/deployment.yaml:
```diff
@@ -323,9 +323,23 @@ spec:
+            kube__healthz_down:
+            Path("/kube-system/healthz") &&
+            Shutdown() &&
+            SourceFromLast("10.0.0.0/8", "192.168.0.0/16", "172.16.0.0/12", "127.0.0.1/8", "fd00::/8", "::1/128"{{- if and (eq .Cluster.Provider "zalando-eks") (eq .Cluster.ConfigItems.eks_ip_family "ipv6")}}{{ range $ip := split .Values.subnet_ipv6_cidrs "," }},"{{ $ip }}"{{ end }}{{- end }})
```
I'm not sure it makes sense to have the ipv6 ranges AND the specific ranges for IPv6 when zalando-eks is configured.

Basically the ranges: `"fd00::/8", "::1/128"` has no meaning in our setup from my understanding.

ponimas replied:
fixed

szuecs replied:
Hmm ::1 should be useful for debugging purposes if you execute into the pod. Don't you think so?

ponimas replied:
But from inside the pods ipv4 loopback should be available?

szuecs replied:
It depends if we have IPv4 enabled or disabled. If we have dualstack enabled we also have IPv4 but it simplifies the network if we would disable IPv4 and maybe it fixes a teapot issue with the svc type load balancer issue if we would disable IPv4.

ponimas replied:
Makes sense, i'll enable the looopback back.

ponimas committed:
fix: Update IPv6 routing rules for EKS clusters

The changes modify IPv6 routing configuration in Skipper deployment for
healthz endpoints. IPv6 LAN routes (fd00::/8 and ::1/128) are now only
included in non-EKS IPv6 environments, while EKS IPv6 configurations
properly use the subnet values from environment variables.

ponimas commented:
👍

ayruslore commented:
👍

ponimas committed:
fix: unconditionally enable the ::1/128 (IPv6 loopback)

Signed-off-by: Aleksandr Ponimaskin <aleksandr.ponimaskin@zalando.de>

ponimas commented:
👍

MustafaSaber commented:
This PR & the following supersedes https://github.com/zalando-incubator/kubernetes-on-aws/pull/8894 and fix pipeline issues there.

MustafaSaber commented:
👍
